rootfs/apply.go

```diff
@@ -69,7 +69,7 @@ func ApplyLayer(snapshots snapshot.Snapshotter, mounter Mounter, rd io.Reader, p
 	digester := digest.Canonical.Digester() // used to calculate diffID.
 	rd = io.TeeReader(rd, digester.Hash())
 
-	if _, err := archive.Apply(context.Background(), key, rd); err != nil {
+	if _, err := archive.Apply(context.Background(), dir, rd); err != nil {
```

Comment:
@stevvooe  Sorry, I did not describe it clearly.
The function `ApplyLayer` need return the `diffID` which is a input parameter of `registerDiffID` and is used to generate the `ChainID` in function `Prepare`.